Services/ProtocolServer/HttpsProtocol.h

```diff
@@ -26,16 +26,20 @@
 
 #pragma once
 
+#include <LibTLS/Certificate.h>
 #include <ProtocolServer/Protocol.h>
 
 namespace ProtocolServer {
 
 class HttpsProtocol final : public Protocol {
 public:
-    HttpsProtocol();
+    HttpsProtocol(const Vector<Certificate>&);
     virtual ~HttpsProtocol() override;
 
     virtual OwnPtr<Download> start_download(ClientConnection&, const String& method, const URL&, const HashMap<String, String>& headers, const ByteBuffer& request_body) override;
+
+private:
+    const Vector<Certificate>& m_ca_certificates;
```

Comment:
This feels a bit sloppy. How about we make a certificate manager that `HttpsProtocol` and `GeminiProtocol` can access through a singleton instance getter (or similar)?

Reply:
For now, I went with something lower-level, a singleton `DefaultRootCACertificates`, and the *Protocol code remains mostly unchanged.
the Jobs just take an optional override param, or default to the...default.

If at some point we decide to allow the user to pick between different sets of root CAs, we'll change to your suggested certificate manager.